lib/Markup/LineList.cpp

```diff
@@ -34,8 +34,11 @@ std::string LineList::str() const {
   auto InitialIndentation = measureIndentation(FirstLine->Text);
 
   for (auto Line = FirstLine; Line != Lines.end(); ++Line) {
+    if (Line != FirstLine) {
```

Comment:
Do you mean the end line here? Also, does the line have equality that just checks the range rather than the text itself? If not, might be better to just compare the range here rather than the text.

Reply:
Ah, true.
```
Stream << FirstLine->Text.drop_front(InitialIndentation);
for (auto *Line : Lines.drop_front()) {
  auto Drop = std::min(InitialIndentation, Line->FirstNonspaceOffset);
  Stream << "\n" << Line->Text.drop_front(Drop);
}
```
Though I didn't check what `Lines` is, so that may also need an `ArrayRef(Lines)`.

Reply:
Good suggestion. Added a variation of it. We still need to use C-style for loop because `FirstLine` might not be the first line, just the first line with content and we need to respect that.

Reply:
Oh that's a confusing name then 😅, thanks for updating. The above could just keep an index instead of bumping the pointer and then you could go for each, but this works too 👍